### DISCUSSION TITLE
How to prevent RadioButton (that are dynamically mounted to RadioSet) from expansion animation on mouse click?

### DISCUSSION THREAD

**Discussion by learnbyexample** [Textualize/textual#5213] [Q&A] [answered]
Here's a sample program:

```python
from textual.app import App
from textual.widgets import RadioButton, RadioSet

class TestApp(App):

    CSS = ''' Screen { align: center middle; } '''

    def __init__(self):
        super().__init__()
        self.rset = RadioSet()    
    
    def compose(self):
        yield RadioSet('apple', 'banana', 'cherry')
        yield self.rset

    def on_mount(self):
        self.dark = False
        rb1 = RadioButton('true')
        self.rset.mount(rb1)
        rb2 = RadioButton('false')
        self.rset.mount(rb2)

if __name__ == "__main__":
    app = TestApp()
    app.run()
```

Here's a sample video when the above program is run:

https://github.com/user-attachments/assets/f2fbabf2-27c7-4320-a0c1-8ac7a2fa3b15

Is there a way to prevent the RadioButtons from animating in the second case (the RadioSet with true/false choices)?

For my actual use case, I need to dynamically add/remove existing RadioButtons from a RadioSet (the number of RadioButtons vary, need to show a pre-selected choice if applicable, change styling, etc - which is why I'm using `mount`).

**Comment by willmcgugan** [comment, marked as answer]
I didn't anticipate anyone adding a RadioButton to a RadioSet after it is mounted. That's probably easy to make that work, you might want to file an issue.

In the meantime, recreating the RadioSet with your new options is probably the way to go.

**Reply by learnbyexample** [reply]
I've opened an issue. I'll also try out your suggestion of recreating RadioSet.

**Reply by learnbyexample** [reply]
I experimented with recreating the RadioSet for my actual project and seems to work fine so far (I still have some scenarios to run through).

One major difference compared to `mount` is that the first option is soft-selected and underlined (`apple` in the screenshot below) - is there a way to prevent this behavior?

![rset_example](https://github.com/user-attachments/assets/c9d06c71-c31c-4cfb-81ef-da8d29f4ff8d)

**Reply by learnbyexample** [reply]
I have a partial workaround by moving the `focus` away from the RadioSet to another widget. I'd prefer to be able to have focus on the RadioSet without the underline, but overall I'm satisfied with the option of recreating RadioSet instead of mounting RadioButtons.
